### ISSUE TITLE
GeographicLocationSerializer fail parsing Lat and Lon on other culture

### ISSUE THREAD

**Issue by dahermansson** [ical-org/ical.net#124]
```
double.TryParse(values[0], out lat);
double.TryParse(values[1], out lon);
```

Fails to convert values like 12.3456 if the current culture use 12,3456 as decimal seperator. 

Use CultureInfo.InvariantCulture to fix? 

```
double.TryParse(values[0], NumberStyles.Any, CultureInfo.InvariantCulture, out lat);
double.TryParse(values[1], NumberStyles.Any, CultureInfo.InvariantCulture, out lon);
```

**Comment by rianjs** [comment]
RFC 5545 specifies that the [GEO](https://tools.ietf.org/html/rfc5545#section-3.8.1.6) component be a [FLOAT](https://tools.ietf.org/html/rfc5545#section-3.3.7). A FLOAT is defined as:

```
Format Definition:  This value type is defined by the following
      notation:

       float      = (["+"] / "-") 1*DIGIT ["." 1*DIGIT]

   Description:  If the property permits, multiple "float" values are
      specified by a COMMA-separated list of values.

      No additional content value encoding (i.e., BACKSLASH character
      encoding, see Section 3.3.11) is defined for this value type.

   Example:

       1000000.0000001
       1.333
       -3.14
```

The definition of a FLOAT doesn't allow for decimal commas, because allowing localized flavors of RFC-defined primitives makes the serialization-deserialization process asymmetric: _going from ical text to object representation and back again should always result in the same inputs and outputs_. Decimal commas would never be an output of the serialization process.

I'm sympathetic to being a little more lenient on parsing ical text, but I think it's a slippery slope: this wouldn't just be an exception on GEO properties, it would be an exception for all FLOATs. And once you make an exception for FLOATs, why not make an exception for the other RFC-defined primitives? Complexity grows quickly, and the value of the spec is diluted.

I think the "right" answer here is to massage the data before handing it to ical. I'm guessing your usage scenario has text data generated from somewhere else. If it's another ical library, then that library has a bug. If it's something else, you may need to write some mapping code to convert localized primitives to [the formats the spec allows for](https://tools.ietf.org/html/rfc5545#section-3.3).

I'm sure this isn't the answer you want to hear, but I think it's the right one for now.

**Comment by dahermansson** [comment]
I think you misunderstood my bug. 

I found this bug when running the Unittests. GeographicLocation1_2 fails because of my CurrentCulture (Swedish). 
The input in this case is `GEO:37.386013;-122.082932` as specified in GeographicLocation1.ics and as it should be, not with decimal commas. 

double.parse() is expecting 37,386013 (with decimal comma) by default if not other is specified when running ical.net library (or any form off dot.net code) on a computer who has for example Swedish Culture where we use decimal commas. 

The serialization would have the same problem (it would write out a value with decimal comma) if it not specify a specific format as it does. 
`g.Latitude.ToString("0.000000")`

**Comment by rianjs** [comment]
Ohhhhhhhhhhhhh, I see. You're saying you're in a decimal comma locale, so parsing doesn't work, because it's expecting a decimal comma instead of a dot.

Well then. That's definitely a bug that should be fixed. I wonder if there are other locations that need to be similarly updated...

**Comment by rianjs** [comment]
OK, it looks like that's the only place that parses `float` or `double`. Everything else is `Enum`, `int`, and the various flavors of `DateTime`.

**Comment by dahermansson** [comment]
A quick search gives that this is the only place were there trying to parse a string to a double so i think this is the only one. And this is the only unittest that fails for me.

**Comment by rianjs** [comment]
Your commit that you had accidentally included in the StringBuilder pull request looked good. Can you open a second pull request and reference this issue in the commit message?

I'll merge both PRs, and publish a new nuget version once that's done.

**Comment by rianjs** [comment]
(I'll also give you commit rights after that.)

**Comment by rianjs** [comment]
Published in nuget 2.2.13 / 3651a7923a771fa2742c4aa2e3c4e00011dd5b45
